### PR TITLE
Add spinner in UserMenu to list pending long running actions

### PR DESCRIPTION
**Pull request by gsouquet**
Fixes vector-im/element-web#15451

A couple of components were migrated to TypeScript along the way too.

Screen Shot 2021-05-24 at 15 04 24

Following @niquewoodhouse and @nadonomy work on the design it was decided to implement a spinner on the User Menu to notify of any long running actions that could be occuring. 
Below is a non exhaustive list of things that could trigger this spinner too 
- downloading keys
- indexing rooms
- ...

### CODE REVIEW

File: src/components/views/elements/TooltipButton.tsx [hunk 1]
```diff
@@ -19,8 +19,16 @@ import React from 'react';
...
+}
+
+interface IState {
+    hover: boolean;
+}
+
 @replaceableComponent("views.elements.TooltipButton")
-export default class TooltipButton extends React.Component {
+export default class TooltipButton extends React.Component<IProps, IState> {
     state = {
```
Comment (jryans) [inline src/components/views/elements/TooltipButton.tsx:32]: Move this into the constructor and use `this.state = { ... }` to match our more usual way of doing this.

File: src/components/views/elements/TooltipButton.tsx [hunk 2]
```diff
@@ -19,8 +19,16 @@ import React from 'react';
...
         hover: false,
     };
```
Comment (jryans) [inline src/components/views/elements/TooltipButton.tsx:34]: Mark the event handlers below as `private` explicitly.

**Review by jryans — CHANGES_REQUESTED** [review]
This great code wise overall, just a few small things to tweak with the TS conversion.

Some further thoughts on our TS code style are in [this doc](https://docs.google.com/document/d/1ldMFqKWqV6AAUSbtxar1Y1fu96d1jF8MSjwcK7q-LWk/edit), in case you haven't seen it yet.

File: src/components/views/elements/TooltipButton.tsx (hunk 1 above)
Comment (t3chguy) [reply src/components/views/elements/TooltipButton.tsx:32]: We have a bunch of these too fwiw

Comment (jryans) [reply src/components/views/elements/TooltipButton.tsx]: Well, I was instructed by @turt2live to convert one over in [another PR](https://github.com/matrix-org/matrix-react-sdk/pull/6076#discussion_r636394699)... I don't have a strong opinion myself, but we should perhaps decide collectively whether we care or not.

**Pull request merged**